### PR TITLE
fix: 修复选中态的描边宽度样式问题

### LINKED ISSUE #1624: 🐛 选中态的描边样式问题
### 🖋 Description

image

选中数据单元格的描边的粗细不一致，而且看起来变细了

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #1624

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| image      | image  |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/cell/base-cell.ts [hunk 1]
```diff
@@ -287,11 +287,14 @@ export abstract class BaseCell<T extends SimpleBBox> extends Group {
...
           if (isNumber(style)) {
+            const { horizontalBorderWidth, verticalBorderWidth } =
+              this.theme.dataCell.cell;
+
             const marginStyle = {
-              x: x + style / 2,
-              y: y + style / 2,
-              width: width - style - 1,
-              height: height - style - 1,
+              x: x + verticalBorderWidth / 2 + style / 2,
```
Comment (wjgogogo) [inline packages/s2-core/src/cell/base-cell.ts:294]: 本质上还是`GridGroup`在绘制边框后，会被置于顶层，所以将hoverFocus边框给盖住了。所以在绘制边框时，需要向内收缩`GridGroup`绘制边框的宽度，避免重叠被遮住

Comment (xingwanying) [reply packages/s2-core/src/cell/base-cell.ts:294]: 加个单测？避免此类重构修改导致样式有问题

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**